apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetService.java

```diff
@@ -266,11 +270,13 @@ public void delete(@NonNull UUID id) {
     }
 
     @Override
-    public DatasetPage find(int page, int size, @NonNull DatasetCriteria criteria) {
+    public DatasetPage find(int page, int size, @NonNull DatasetCriteria criteria, String sorting) {
         String workspaceId = requestContext.get().getWorkspaceId();
         String userName = requestContext.get().getUserName();
         String workspaceName = requestContext.get().getWorkspaceName();
 
+        String sortingFields = sortingQueryBuilder.toOrderBySql(sortingFactory.newSorting(sorting));
+
```

Comment:
I moved `sortingFactory` to controller level.
Regarding `sortingQueryBuilder.toOrderBySql`,  I left it here, since in this case DAO is MySQL DAO which is just an interface without any logic. Also similar approach is used for ProjectService.find